### DISCUSSION TITLE
Why does Nuxt/Vite bundle the same library version twice under some circumstances?

### DISCUSSION THREAD

**Discussion by fabis94** [nuxt/nuxt#20815] [Questions]
I've had this issue with prosemirror and vee-validate where the packages didn't work correctly, because they relied on some sort of global state or Symbols that were bundled twice.

I fixed the issue by adding the packages to `vite.resolve.dedupe` in the nuxt config, but why do I need to do this in the first place? Why would it bundle a library twice?

In my case I think it happened because I have a monorepo with a UI component package and a nuxt app. Nuxt app has vee-validate installed and the UI component package has vee-validate installed as well. And I'm guessing Nuxt got confused when it saw vee-validate at 2 different locations and bundled them twice.

**Comment by warflash** [comment]
Have you made sure that the vee-validate package version is exactly the same in your UI package and your nuxt app?

**Reply by fabis94** [reply]
Yes, i've double checked that it's the exact same version in both `node_modules` folders
